compiler/rustc_hir_typeck/messages.ftl

```diff
@@ -75,3 +75,9 @@ hir_typeck_union_pat_dotdot = `..` cannot be used in union patterns
 
 hir_typeck_arg_mismatch_indeterminate = argument type mismatch was detected, but rustc had trouble determining where
     .note = we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new
+
+hir_typeck_suggest_boxing_note = for more on the distinction between the stack and the heap, see:
+    https://doc.rust-lang.org/book/ch15-01-box.html, https://doc.rust-lang.org/rust-by-example/std/box.html, and
+    https://doc.rust-lang.org/std/boxed/index.html
```

Comment:
Let's keep the old line wrapping, not convinced the new one is that much easier to parse.

```suggestion
hir_typeck_suggest_boxing_note = for more on the distinction between the stack and the heap, see: \
    https://doc.rust-lang.org/book/ch15-01-box.html, https://doc.rust-lang.org/rust-by-example/std/box.html, and \
    https://doc.rust-lang.org/std/boxed/index.html
```

Reply:
That is originally what i had, it doesn't actually cause the line to wrap, instead it will ouput this:
```
   = note: for more on the distinction between the stack and the heap, see: \
           https://doc.rust-lang.org/book/ch15-01-box.html, https://doc.rust-lang.org/rust-by-example/std/box.html, and \
           https://doc.rust-lang.org/std/boxed/index.html
help: try storing this on the heap by calling `Box::new`
```
and i think it looks a bit nicer the way it is now, but it also seems fine? it's also done like 20 lines up for another diagnostic.
https://github.com/rust-lang/rust/blob/b4571bed994a901e6dbd0b28e25e91a57758c549/compiler/rustc_hir_typeck/messages.ftl#L60-L64

Reply:
I can put it all on the one line, that way it would wrap like before?

Reply:
Not sure if understand what you mean by "it's also done lines 20 lines up" -- that `hir_typeck_fru_suggestion` diagnostic doesn't (or... shouldn't) have any newlines in it.

Reply:
it.. appears to have newlines to me, it even has one directly after the `=`, but like i said, i can but it all on the one line that way it should wrap like before i think. would that be better?

Reply:
The newlines there split up the Fluent branches but don't actually appear in the output

Reply:
Ahh, okay that does make more sense, thank you. I didn't find the fluent syntax guide up until a few minutes ago (originally i was just reading the [translation](https://rustc-dev-guide.rust-lang.org/diagnostics/translation.html) and the [diagnostic structs](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-structs.html) pages), specifically this page explains what `hir_typeck_fru_suggestion` is doing: https://projectfluent.org/fluent/guide/selectors.html, which definitely makes more sense now.

But to make `hir_typeck_suggest_boxing_note` easier to parse and also to make the line wrap like before, i can put it all on the one line, should i do that?

Reply:
Sure, do that please

Reply:
done.